core/sql/parser/SqlParserAux.cpp

```diff
@@ -863,8 +863,7 @@ ItemExpr *literalOfNumericPassingScale(NAString *strptr, char sign,
   } else if (strSize <= 19) {
     datatype = (createSignedDatatype ? REC_BIN64_SIGNED : REC_BIN64_UNSIGNED);
     length = sizeof(Int64);
-  } else if (strSize == 20) {
-    createSignedDatatype = FALSE;
+  } else if ((strSize == 20) && (!createSignedDatatype)) {
```

Comment:
This is in the surround, but I wonder whether we shouldn't also look at the TRAF_LARGEINT_UNSIGNED_IO CQD and only generate unsigned largeint literals if that CQD is set to ON.